src/platform/plugins/shared/discover/public/application/main/state_management/redux/internal_state.ts

```diff
@@ -287,6 +287,41 @@ export const internalStateSlice = createSlice({
         tab.uiState.fieldList = action.payload.fieldListUiState;
       }),
 
+    setFieldListExistingFieldsInfoUiState: (
+      state,
+      action: TabAction<{
+        fieldListExistingFieldsInfo: TabState['uiState']['fieldListExistingFieldsInfo'];
+      }>
+    ) =>
+      withTab(state, action, (tab) => {
+        tab.uiState.fieldListExistingFieldsInfo = action.payload.fieldListExistingFieldsInfo;
+      }),
+
+    resetAffectedFieldListExistingFieldsInfoUiState: (
+      state,
+      action: PayloadAction<{
+        dataViewId: string;
+      }>
+    ) => {
+      state.tabs.byId = Object.fromEntries(
+        Object.entries(state.tabs.byId).map(([tabId, tab]) => {
+          if (tab.uiState.fieldListExistingFieldsInfo?.dataViewId === action.payload.dataViewId) {
+            return [
+              tabId,
+              {
+                ...tab,
+                uiState: {
+                  ...tab.uiState,
+                  fieldListExistingFieldsInfo: undefined,
+                },
+              },
+            ];
+          }
+          return [tabId, tab];
+        })
+      );
```

Comment:
```suggestion
      Object.values(state.tabs.byId).forEach((tab) => {
        if (tab.uiState.fieldListExistingFieldsInfo?.dataViewId === action.payload.dataViewId) {
          tab.uiState.fieldListExistingFieldsInfo = undefined;
        }
      });
```
Nit: This works fine, but you could make it a lot simpler by mutating the state with Immer.